onmt/utils/loss.py

```diff
@@ -178,15 +179,27 @@ def _stats(self, loss, scores, target):
             loss (:obj:`FloatTensor`): the loss computed by the loss criterion.
             scores (:obj:`FloatTensor`): a score for each possible output
             target (:obj:`FloatTensor`): true targets
+            batch_sz (:'int'): batch size
 
         Returns:
             :obj:`onmt.utils.Statistics` : statistics for this batch.
         """
-        pred = scores.max(1)[1]
-        non_padding = target.ne(self.padding_idx)
-        num_correct = pred.eq(target).masked_select(non_padding).sum().item()
+        pred = scores.max(2)[1]
+        precision_matches, precision_totals, \
+            prediction_lengths, reference_lengths = \
+            bleu_calculations(
+                    pred,
+                    target,
+                    4,
+                    {self.padding_idx})
+
```

Comment:
remove, again loss.py should not change

Reply:
I am not able to not change loss.py at all and achieve this.